v2/bigtable-cdc-to-hbase/src/main/java/com/google/cloud/teleport/v2/templates/transforms/ConvertChangeStream.java

```diff
@@ -0,0 +1,287 @@
+      if (enabled) {
+        checkArgument(cbtQualifierInput != null, "cbt qualifier cannot be null.");
+        checkArgument(hbaseQualifierInput != null, "hbase qualifier cannot be null.");
+      }
+      twoWayReplication = enabled;
+      cbtQualifier = cbtQualifierInput;
+      hbaseQualifier = hbaseQualifierInput;
+    }
+
+    private boolean twoWayReplication;
+    private String cbtQualifier;
+    private String hbaseQualifier;
+
+    @Override
+    public PCollection<KV<byte[], RowMutations>> expand(
+        PCollection<KV<ByteString, ChangeStreamMutation>> input) {
+      return input.apply(
+          ParDo.of(
+              new ConvertChangeStreamMutationFn(twoWayReplication, cbtQualifier, hbaseQualifier)));
+    }
+  }
+
+  /** Converts Bigtable change stream mutations to Hbase RowMutations objects. */
+  public static class ConvertChangeStreamMutationFn
+      extends DoFn<KV<ByteString, ChangeStreamMutation>, KV<byte[], RowMutations>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConvertChangeStreamMutationFn.class);
+
+    private String hbaseQualifier;
+    private String cbtQualifier;
+    private boolean twoWayReplication;
+
+    public ConvertChangeStreamMutationFn(
+        boolean twoWayReplicationFlag, String cbtQualifierInput, String hbaseQualifierInput) {
+      twoWayReplication = twoWayReplicationFlag;
+      hbaseQualifier = hbaseQualifierInput;
+      cbtQualifier = cbtQualifierInput;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) throws Exception {
+      ChangeStreamMutation mutation = c.element().getValue();
+      // Skip element if it was replicated from HBase.
+      if (twoWayReplication && isHbaseReplicated(mutation, hbaseQualifier)) {
+        return;
+      }
+      RowMutations hbaseMutations = convertToRowMutations(mutation);
+      // Append origin information to mutations.
+      if (twoWayReplication) {
+        appendSourceTagToMutations(hbaseMutations, cbtQualifier);
+      }
+      c.output(KV.of(hbaseMutations.getRow(), hbaseMutations));
+    }
+
+    /**
+     * Checks if mutation was replicated from HBase.
+     *
+     * @param mutation from change stream
+     * @return true if mutation was replicated from hbase
+     */
+    private boolean isHbaseReplicated(ChangeStreamMutation mutation, String hbaseQualifierInput) {
+      List<Entry> mutationEntries = mutation.getEntries();
+
+      if (mutationEntries.size() == 0) {
+        return false;
+      }
+      Entry lastEntry = mutationEntries.get(mutationEntries.size() - 1);
+
+      if (lastEntry instanceof DeleteCells) {
+        if (((DeleteCells) lastEntry)
+            .getQualifier()
+            .equals(ByteString.copyFromUtf8(hbaseQualifierInput))) {
+          Metrics.counter("HbaseRepl", "mutations_filtered_from_hbase").inc();
+          return true;
+        }
+      }
+      Metrics.counter("HbaseRepl", "mutations_replicated_from_bigtable").inc();
```

Comment:
What is this parameter "HbaseRepl", how is this used, should we use more verbose name?

Reply:
Per https://beam.apache.org/releases/javadoc/2.1.0/org/apache/beam/sdk/metrics/Metrics.html guidelines, swapped "HbaseRepl" out for ConvertChnageStreamMutationFn